### PR TITLE
Add a test to ensure upgrades work

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

**Comment by faisal-memon** [comment]
Thanks @kfox1111 or this submission.

File: .github/workflows/helm-chart-ci.yaml [hunk 1]
```diff
@@ -239,3 +239,44 @@ jobs:
 
       - name: Install and test example
         run: ${{ matrix.example }}/run-tests.sh
+
+  upgrade-test:
+    runs-on: ubuntu-22.04
```
Comment (faisal-memon) [inline .github/workflows/helm-chart-ci.yaml:244]: Do we need a stub in helm-chart-ci-ignore?

Comment (kfox1111) [reply .github/workflows/helm-chart-ci.yaml:244]: Good question. Not sure. Those are only for blocking tests? Should we make this blocking too or wait  for a bit to see how well it works?

Comment (faisal-memon) [reply .github/workflows/helm-chart-ci.yaml:244]: Good point. Not going to make this blocking for now.

File: examples/production/run-tests.sh [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 set -xe
 
+UPGRADE_VERSION=v0.13.0
+UPGRADE_REPO=https://spiffe.github.io/helm-charts
```
Comment (faisal-memon) [inline examples/production/run-tests.sh:6]:
```suggestion
UPGRADE_REPO=https://spiffe.github.io/helm-charts-hardened
```

File: examples/production/run-tests.sh [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 set -xe
 
+UPGRADE_VERSION=v0.13.0
```
Comment (faisal-memon) [inline examples/production/run-tests.sh:5]: Is this intended to be the latest version of the chart?

File: examples/production/run-tests.sh (hunk 2 above)
Comment (kfox1111) [reply examples/production/run-tests.sh:6]: Good catch

File: examples/production/run-tests.sh (hunk 3 above)
Comment (kfox1111) [reply examples/production/run-tests.sh:5]: ultimately I think? Was thinking we'd bump it on release for now, and work on an auto detection script that would set it to whatever is latest automatically in a future pr?

Comment (faisal-memon) [reply examples/production/run-tests.sh:5]: haha read my mind with the auto detection.

**Review by faisal-memon — APPROVED** [review]

**Comment by faisal-memon** [comment]
@kfox1111 Just need dco and we are good to go.

**Comment by kfox1111** [comment]
Fixed.

I think we lost the option to auto signoff on merged suggestions.

**Comment by faisal-memon** [comment]
> Fixed.
> 
> I think we lost the option to auto signoff on merged suggestions.

Yes, will fix later today.

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**